nilearn/maskers/surface_masker.py

```diff
@@ -150,23 +150,27 @@ def _fit_mask_img(self, img):
         ----------
         img : SurfaceImage object or :obj:`list` of SurfaceImage or None
         """
-        if self.mask_img is not None:
-            if img is not None:
-                check_same_n_vertices(self.mask_img.mesh, img.mesh)
-            self.mask_img_ = self.mask_img
-            return
-
         if img is None:
-            raise ValueError(
-                "Please provide either a mask_img "
-                "when initializing the masker "
-                "or an img when calling fit()."
-            )
+            if self.mask_img is None:
+                raise ValueError(
+                    "Please provide either a mask_img "
+                    "when initializing the masker "
+                    "or an img when calling fit()."
+                )
+
+            if self.mask_img is not None:
+                self.mask_img_ = self.mask_img
+                return
 
         if not isinstance(img, list):
             img = [img]
         img = concatenate_surface_images(img)
 
+        if self.mask_img is not None:
+            check_same_n_vertices(self.mask_img.mesh, img.mesh)
+            self.mask_img_ = self.mask_img
+            return
+
```

Comment:
I feel like this could be combined with the if block above?

Reply:
how so?

Reply:
ok maybe not just had a _feeling_ 😝

Reply:
# 🤣